bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoCrockpotHandler.java

```diff
@@ -0,0 +1,323 @@
+
+        this.wemoHttpCaller = wemohttpCaller;
+
+        logger.debug("Creating a WemoCrockpotHandler for thing '{}'", getThing().getUID());
+
+        if (upnpIOService != null) {
+            this.service = upnpIOService;
+        } else {
+            logger.debug("upnpIOService not set.");
+        }
+    }
+
+    @Override
+    public void initialize() {
+        Configuration configuration = getConfig();
+
+        if (configuration.get("udn") != null) {
+            logger.debug("Initializing WemoCrockpotHandler for UDN '{}'", configuration.get("udn"));
+            service.registerParticipant(this);
+            onSubscription();
+            onUpdate();
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            logger.debug("Cannot initalize WemoCrockpotHandler. UDN not set.");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("WeMoCrockpotHandler disposed.");
+
+        removeSubscription();
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("Command '{}' received for channel '{}'", command, channelUID);
+        String mode = "0";
+        String time = null;
+
+        if (command instanceof RefreshType) {
+            try {
+                updateWemoState();
+            } catch (Exception e) {
+                logger.debug("Exception during poll", e);
+            }
+        } else if (channelUID.getId().equals(CHANNEL_COOKMODE)) {
+            String commandString = command.toString();
+            switch (commandString) {
+                case "OFF":
+                    mode = "0";
+                    time = "0";
+                    break;
+                case "WARM":
+                    mode = "50";
+                    break;
+                case "LOW":
+                    mode = "51";
+                    break;
+                case "HIGH":
+                    mode = "52";
+                    break;
+            }
+            try {
+                String soapHeader = "\"urn:Belkin:service:basicevent:1#SetBinaryState\"";
+                String content = "<?xml version=\"1.0\"?>"
+                        + "<s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\" s:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">"
+                        + "<s:Body>" + "<u:SetCrockpotState xmlns:u=\"urn:Belkin:service:basicevent:1\">" + "<mode>"
+                        + mode + "</mode>" + "<time>" + time + "</time>" + "</u:SetCrockpotState>" + "</s:Body>"
+                        + "</s:Envelope>";
+                String wemoURL = getWemoURL("basicevent");
+
+                if (wemoURL != null) {
+                    wemoHttpCaller.executeCall(wemoURL, soapHeader, content);
+                }
+            } catch (Exception e) {
```

Comment:
If you catch all `Exception` you are potentially also catching for example `NullPointerException`s and  they tend to miss a message, so you will be printing the message `null` which will make it troublesome to debug, in this case it is always better to log the complete stacktrace.

Reply:
Don't get your point here. Could you give me an example please.

Reply:
Catching all `Exception`'s is not a best practice because it 'handles' also coding errors in the same way as `IOException`'s and such. If for example, `wemoHttpCaller` could become `null` somehow because of a programming error, a `NullPointerException` would be thrown, the NPE and others have the message unset and thus returning `null` when requesting the message to be logged/reported to the user.

If possible it would be way better catch only specific exceptions, maybe `IOException`'s or whatever exception will be thrown by the main code.

Reply:
Done